test/jdk/java/util/zip/DeflaterDictionaryTests.java

```diff
@@ -0,0 +1,185 @@
+import java.util.zip.Deflater;
+import java.util.zip.Inflater;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * @test
+ * @bug 8252739
+ * @summary Verify Deflater.setDictionary(dictionary, offset, length) uses the offset
+ * @run testng/othervm DeflaterDictionaryTests
+ */
+public class DeflaterDictionaryTests {
+    // Output buffer size
+    private static final int RESULT_SIZE = 1024;
+    // Data to compress
+    private static final String SRC_DATA = "Welcome to US Open;"
+            +"Welcome to US Open;"
+            +"Welcome to US Open;"
+            +"Welcome to US Open;"
+            +"Welcome to US Open;"
+            +"Welcome to US Open;";
+    // Dictionary to be used
+    private static final String DICTIONARY = "US Open";
+
+    /**
+     * Validate that an offset can be used with Deflater::setDictionary
+     * @throws Exception if an error occurs
+     */
+    @Test
+    public void ByteArrayTest() throws Exception {
+        byte[] input = SRC_DATA.getBytes(UTF_8);
+        byte[] output = new byte[RESULT_SIZE];
+        // Compress the bytes
+        Deflater deflater = new Deflater();
+        deflater.setDictionary(DICTIONARY.getBytes(UTF_8), 1, 3);
+        deflater.setInput(input);
+        deflater.finish();
+        int compressedDataLength = deflater.deflate(output,0 , output.length, Deflater.NO_FLUSH);
+        System.out.printf("Deflater::getTotalOut:%s, Deflater::getAdler: %s," +
+                        " compressed length: %s%n", deflater.getTotalOut(),
+                deflater.getTotalOut(), compressedDataLength);
+        deflater.finished();
+
+        // Decompress the bytes
+        Inflater inflater = new Inflater();
+        inflater.setInput(output, 0, compressedDataLength);
+        byte[] result = new byte[RESULT_SIZE];
+        int resultLength = inflater.inflate(result);
+        if(inflater.needsDictionary()) {
+            System.out.println("Specifying Dictionary");
+            inflater.setDictionary(DICTIONARY.getBytes(UTF_8), 1, 3);
+            resultLength = inflater.inflate(result);
+        } else {
+            System.out.println("Did not need to use a Dictionary");
+        }
+        inflater.finished();
+        System.out.printf("Inflater::getAdler:%s, length: %s%n",
+                inflater.getAdler(), resultLength);
+
+        Assert.assertEquals(SRC_DATA.length(), resultLength);
+        Assert.assertEquals(input, Arrays.copyOf(result, resultLength));
+
+        // Release Resources
+        deflater.end();
+        inflater.end();
+    }
+
+    /**
+     * Validate that a ByteBuffer can be used with Deflater::setDictionary
+     * @throws Exception if an error occurs
+     */
+    @Test
+    public void testByteBufferWrap() throws Exception {
+        byte[] input = SRC_DATA.getBytes(UTF_8);
+        byte[] output = new byte[RESULT_SIZE];
+        // Compress the bytes
+        ByteBuffer dictDef = ByteBuffer.wrap(DICTIONARY.getBytes(UTF_8));
```

Comment:
This should use a slice of the byte buffer, so array offset is not 0. This can be done with offset and size on the wrap() call.